client/src/components/Library/Allbooks.jsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-unused-vars */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { getBooks, setPagination } from '../../actions/bookAction';
+import Book from '../Library/Book.jsx';
+import generic from '../../assets/images/generic.jpg';
+
+
+/** component that displays all books
+ * @class Allbooks
+ * @extends {Component}
+ */
+class Allbooks extends Component {
+  /** Creates an instance of Allbooks.
+ * @param {any} props
+ * @memberof Allbooks
+ */
+  constructor(props) {
+    super(props);
+    this.state = {
+      books: {},
+      limit: 8,
+      offset: 0,
+      pages: []
+    };
+    this.getPages = this.getPages.bind(this);
+    this.getNewPage = this.getNewPage.bind(this);
+  }
+  /** @returns {*} void
+  * @memberof Allbooks
+  */
+  componentWillMount() {
+    this.props.getBooks(this.state.limit, this.state.offset);
+  }
+
+  /** @description displays pagination
+   * @returns {*} void
+   * @param {any} nextProps
+   * @memberof Allbooks
+   */
+  componentWillReceiveProps(nextProps) {
+    if (this.props.pagination.pageCount !== nextProps.pagination.pageCount) {
+      this.getPages(nextProps.pagination.pageCount);
+    }
+  }
+  /** @description creates an array of page numbers
+   * @returns {*} void
+   * @param {any} pageCount
+   * @memberof Allbooks
+   */
+  getPages(pageCount) {
```

Comment:
What is this method for?

Reply:
it returns an array of the page numbers for the pagination component.